### PR TITLE
feat: init server project

### PR DESCRIPTION
**Pull request by IRONICBo**
#### 🔍 What type of PR is this?

/kind feature

#### 👀 What this PR does / why we need it:

init server project

#### 🅰 Which issue(s) this PR fixes:

issue: https://github.com/OpenIMSDK/OpenKF/issues/3

Fixes # Add server base project.

### CODE REVIEW

File: server/internal/config/viper.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/spf13/viper"
+)
+
+func init() {
+	configPath := "./config.yaml"
+	fmt.Printf("[OpenKF] Load config from %s ... ", configPath)
+	viper.SetConfigFile(configPath)
+
+	if err := viper.ReadInConfig(); err != nil {
+		panic(fmt.Errorf("fatal error config file: %s", err))
```
Comment (cubxxw) [inline server/internal/config/viper.go:19]: log record?

File: server/internal/config/viper.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+package config
+
+import (
+	"fmt"
+
+	"github.com/spf13/viper"
+)
+
+func init() {
+	configPath := "./config.yaml"
```
Comment (cubxxw) [inline server/internal/config/viper.go:14]: hard coded ~

File: server/main.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	"github.com/OpenIMSDK/OpenKF/server/pkg/log"
+	"github.com/OpenIMSDK/OpenKF/server/pkg/server"
+)
+
+func init() {
+	// init config
+	config.ConfigInit()
+
+	// logo
+	_logo := `
```
Comment (cubxxw) [inline server/main.go:22]: Can you can, go to https://github.com/avelino/awesome-go to find more pattern, command line beautification output what ha ha ha ha ha

File: server/pkg/log/fileline_hook.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
+// Copyright © 2023 OpenIMSDK open source community. All rights reserved.
+// Licensed under the MIT License (the "License");
+// you may not use this file except in compliance with the License.
+
+package log
```
Comment (cubxxw) [inline server/pkg/log/fileline_hook.go:5]: logrus vs zap

File: server/config.yaml
Comment (AllianceTing) [inline server/config.yaml:1]: I suggest implementing an authentication mechanism in Redis to enhance the security of its connections.

File: server/config-example.yaml
Comment (AllianceTing) [inline server/config-example.yaml:1]: I suggest implementing an authentication mechanism in Redis to enhance the security of its connections.

File: server/config.yaml
Comment (AllianceTing) [inline server/config.yaml:1]: I suggest implementing an authentication mechanism in Redis to enhance the security of its connections.

File: server/internal/config/viper.go (hunk 2 above)
Comment (IRONICBo) [reply server/internal/config/viper.go:14]: I need to specify a path to read the configuration file, do I need to consider reading the configuration file from the command line?

File: server/internal/client/minio.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+// Licensed under the MIT License (the "License");
+// you may not use this file except in compliance with the License.
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"io"
+
```
Comment (AllianceTing) [inline server/internal/client/minio.go:11]: there is one more spacce here

File: server/config.yaml
Comment (IRONICBo) [reply server/config.yaml:1]: Ok, I will update it.

File: server/main.go (hunk 3 above)
Comment (IRONICBo) [reply server/main.go:22]: Sounds good, I will try to choose one.

**Review by AllianceTing — COMMENTED** [review]
agree to merge

**Pull request merged**